source/winAPI/messageWindow.py

```diff
@@ -4,19 +4,152 @@
 # See the file COPYING for more details.
 
 """
-A submodule for NVDA's message window, used for handling Windows Messages.
+A submodule for NVDA's message window, used for handling Window Messages.
 
 Message windows can be used to handle communications from other processes, new NVDA instances and Windows.
 """
 
-import enum
+from ctypes.wintypes import HWND
+from enum import IntEnum
+from typing import (
+	Optional,
+)
 
+import wx
 
-class WindowMessage(enum.IntEnum):
-	POWERBROADCAST = 0x0218
-	WTSSESSION_CHANGE = 0x02B1
+from . import (
+	displayTracking,
+	powerTracking,
+	sessionTracking,
+)
+import extensionPoints
+import gui
+import windowUtils
+
+
+post_windowMessageReceipt = extensionPoints.Action()
+"""
+Notifies when a window message has been received by NVDA.
+This allows components to perform an action when several system events occur,
+such as power, screen orientation and hardware changes.
+
+Handlers are called with three arguments.
+@param msg: The window message.
+@type msg: int
+@param wParam: Additional message information.
+@type wParam: int
+@param lParam: Additional message information.
+@type lParam: int
+"""
+
+
+class WindowMessage(IntEnum):
+	DISPLAY_CHANGE = 0x7e
 	"""
+	WM_DISPLAYCHANGE
+
+	https://docs.microsoft.com/en-us/windows/win32/gdi/wm-displaychange
+	"""
+	POWER_BROADCAST = 0x0218
+	"""
+	WM_POWERBROADCAST
+
+	An application should process this event by calling the GetSystemPowerStatus function
+	to retrieve the current power status of the computer.
+	In particular, the application should check the
+	ACLineStatus, BatteryFlag, BatteryLifeTime, and BatteryLifePercent members
+	of the SYSTEM_POWER_STATUS structure for any changes.
+
+	This event can occur when battery life drops to less than 5 minutes,
+	or when the percentage of battery life drops below 10 percent,
+	or if the battery life changes by 3 percent.
+
+	https://docs.microsoft.com/en-us/windows/win32/power/pbt-apmpowerstatuschange
+	"""
+
+	WTS_SESSION_CHANGE = 0x02B1
+	"""
+	WM_WTSSESSION_CHANGE
+
 	Windows Message for when a Session State Changes.
-	Receiving these messages is registered by registerSessionNotification.
+	Receiving these messages is registered by sessionTracking.register.
 	handleSessionChange handles these messages.
+
+	https://docs.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtsregistersessionnotification
+	"""
+
+
+class _MessageWindow(windowUtils.CustomWindow):

```

Comment:
It could be worth looking at windowUtils also (as a candidate for winAPI) in the future.